app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyElement.java

```diff
@@ -116,7 +125,8 @@ public boolean equals(Object obj) {
                 Objects.equals(order, this.order) &&
                 Objects.equals(surveyCompoundKey, this.surveyCompoundKey) &&
                 Objects.equals(type, other.type) &&
-                Objects.equals(rules, other.rules);
+                Objects.equals(afterRules, other.afterRules) &&
+                Objects.equals(beforeRules, other.beforeRules);
```

Comment:
I'm going to remove the hashCode/equals methods from the SurveyElement class hierarchy. It's broken, and isn't used, even for tests. Creating hashCode/equals methods for SurveyElement which has to be mutable for Dynamo, and subclassable in a way that changes equality, and includes JsonNode (which has abstract methods that need prefab values)... it's just a quagmire and not worth it. So, removed.